pulsar-broker-common/src/main/java/org/apache/pulsar/broker/web/GzipHandlerUtil.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.broker.web;
+
+import java.util.List;
+import org.eclipse.jetty.server.Handler;
+import org.eclipse.jetty.server.handler.gzip.GzipHandler;
+
+public class GzipHandlerUtil {
+    public static Handler wrapWithGzipHandler(Handler innerHandler, List<String> gzipCompressionExcludedPaths) {
+        Handler wrappedHandler;
+        if (isGzipCompressionCompletelyDisabled(gzipCompressionExcludedPaths)) {
```

Comment:
Just thinking out loud - can't we just define gzip on the handlers we see fit? We know Pulsar best no? Too much configuration option can be confusing. They can control the gzip by using headers, no?

Reply:
> can't we just define gzip on the handlers we see fit?

Which handlers would you exclude? I think that it's useful to enable it globally since the current assumption is that all usages will benefit from gzip compression.

> Too much configuration option can be confusing. They can control the gzip by using headers, no?

That's true, but there's risk that regressions are caused when changes are made. Adding the `httpServerGzipCompressionExcludedPaths` is a way to ensure that users aren't stuck if there's a regression for some corner case.
There's a mailing list discussion https://lists.apache.org/thread/xbs1f5v7865fz1hm1vxqlw6o9dgy3d9z .